components/common/video/player.tsx

```diff
@@ -0,0 +1,47 @@
+import Hls from 'hls.js/dist/hls.light' // Use light build of hls.
+import { useEffect, useRef, useState } from 'react'
+
+export default function VideoPlayer({
+  source,
+  poster,
+}: {
+  source: string
+  poster: string
+}) {
+  const [hls] = useState(() => new Hls())
+
+  const ref = useRef<HTMLVideoElement>(null)
+
+  useEffect(() => {
+    if (ref.current && Hls.isSupported()) {
+      hls.attachMedia(ref.current)
+
+      return () => {
+        hls.detachMedia()
+      }
+    }
+  }, [hls])
+
+  useEffect(() => {
+    if (ref.current) {
+      if (Hls.isSupported()) {
+        hls.loadSource(source)
+      } else {
+        // TODO: fallback
```

Comment:
Is a fallback still needed here?

Reply:
yeah would be nicer to just show a fallback message on unsupported browsers. added that in just now.